source/patterns/@aws-solutions-constructs/core/lib/dynamodb-table-helper.ts

```diff
@@ -58,9 +58,11 @@ export interface BuildDynamoDBTableWithStreamProps {
 
 export function buildDynamoDBTable(scope: Construct, props: BuildDynamoDBTableProps): [dynamodb.ITable, dynamodb.Table?] {
   // Conditional DynamoDB Table creation
-  if (props.existingTableObj) {
+  if (props.existingTableInterface && props.existingTableObj) {
+    throw new Error("Must provide either existingTableInterface or existingTableObj");
+  } else if (props.existingTableObj && !props.dynamoTableProps) {
     return [props.existingTableObj, props.existingTableObj];
-  } else if (props.existingTableInterface) {
+  } else if (props.existingTableInterface && !props.dynamoTableProps) {
```

Comment:
We've mixed confirming the inputs with the functionality of routine and hidden what's going on in the process. Leave the code in the function as it was and check the arguments first. You have included checking for props, that's a good thing, i had neglected to include that. It makes looking for 1 and only 1 arg being supplied more complicated though. Maybe something like:

```
let number_of_definitions = 0;
if (props.existingTableObj) { number_of_definitions++; }
if (props.existingTableInterface) { number_of_definitions++; }
if (props.dynamoTableProps) { number_of_definitions++; }
if (number_of_definitions != 1) {
  throw Error('Invalid arguments provided');
}

// rest of function in previous form
```